core/templates/dev/head/domain/utilities/browser-checker.service.ts

```diff
@@ -18,15 +18,19 @@
 
 import { downgradeInjectable } from '@angular/upgrade/static';
 import { Injectable } from '@angular/core';
+import { WindowRef } from
+  'services/contextual/window-ref.service';
 
 const CONSTANTS = require('constants.ts');
 
 @Injectable({
   providedIn: 'root'
 })
 export class BrowserCheckerService {
+  constructor(private windowRef: WindowRef) {}
+
   private _supportsSpeechSynthesis(): boolean {
-    if (window.hasOwnProperty('speechSynthesis')) {
+    if (this.windowRef.nativeWindow.hasOwnProperty('speechSynthesis')) {
```

Comment:
I don't know if this is the right thing to do... @bansalnitish @ankita240796 @kevinlee12 can you please advice?

Reply:
I’d have to defer depending on what’s the best practice for Angular.

Reply:
This is perfect :).

`this.windowRef.nativeWindow` is the correct replacement of window in Angular.